util/data_stream.go

```diff
@@ -55,6 +58,17 @@ type DataStream struct {
 	BasePath string `json:"-" yaml:"-"`
 }
 
+type BaseDataStream struct {
```

Comment:
I was wondering why is it a "base" data stream. I got the idea, but can't relate this name with meaning.